registry/storage/driver/middleware/cloudfront/s3filter_test.go

```diff
@@ -0,0 +1,374 @@
+	AssertEqual(t, false, ips.contains(net.ParseIP("192.169.0.0")))
+}
+
+func TestMatchIPV4WithRegionMatched(t *testing.T) {
+	t.Parallel()
+	server := setupTest(aWSIPResponse{
+		Prefixes: []prefixEntry{
+			{
+				IPV4Prefix: "192.168.0.0/24",
+				Region:     "us-east-1",
+			},
+		},
+	})
+	defer server.Close()
+
+	ips := newAWSIPs(serverIPRanges(server), time.Hour, []string{"us-east-1"})
+	ips.tryUpdate()
+	AssertEqual(t, true, ips.contains(net.ParseIP("192.168.0.0")))
+	AssertEqual(t, true, ips.contains(net.ParseIP("192.168.0.1")))
+	AssertEqual(t, false, ips.contains(net.ParseIP("192.169.0.0")))
+}
+
+func TestMatchIPV4WithRegionMatch_2(t *testing.T) {
+	t.Parallel()
+	server := setupTest(aWSIPResponse{
+		Prefixes: []prefixEntry{
+			{
+				IPV4Prefix: "192.168.0.0/24",
+				Region:     "us-east-1",
+			},
+		},
+	})
+	defer server.Close()
+
+	ips := newAWSIPs(serverIPRanges(server), time.Hour, []string{"us-west-2", "us-east-1"})
+	ips.tryUpdate()
+	AssertEqual(t, true, ips.contains(net.ParseIP("192.168.0.0")))
+	AssertEqual(t, true, ips.contains(net.ParseIP("192.168.0.1")))
+	AssertEqual(t, false, ips.contains(net.ParseIP("192.169.0.0")))
+}
+
+func TestMatchIPV4WithRegionNotMatched(t *testing.T) {
+	t.Parallel()
+	server := setupTest(aWSIPResponse{
+		Prefixes: []prefixEntry{
+			{
+				IPV4Prefix: "192.168.0.0/24",
+				Region:     "us-east-1",
+			},
+		},
+	})
+	defer server.Close()
+
+	ips := newAWSIPs(serverIPRanges(server), time.Hour, []string{"us-west-2"})
+	ips.tryUpdate()
+	AssertEqual(t, false, ips.contains(net.ParseIP("192.168.0.0")))
+	AssertEqual(t, false, ips.contains(net.ParseIP("192.168.0.1")))
+	AssertEqual(t, false, ips.contains(net.ParseIP("192.169.0.0")))
+}
+
+func TestInvalidData(t *testing.T) {
+	t.Parallel()
+	// Invalid entries from aws should be ignored.
+	server := setupTest(aWSIPResponse{
+		Prefixes: []prefixEntry{
+			{IPV4Prefix: "9000"},
+			{IPV4Prefix: "192.168.0.0/24"},
+		},
+	})
+	defer server.Close()
+
+	ips := newAWSIPs(serverIPRanges(server), time.Hour, nil)
+	ips.tryUpdate()
+	AssertEqual(t, 1, len(ips.ipv4))
+}
+
+func TestInvalidNetworkType(t *testing.T) {
+	t.Parallel()
+	server := setupTest(aWSIPResponse{
+		Prefixes: []prefixEntry{
+			{IPV4Prefix: "192.168.0.0/24"},
+		},
+		V6Prefixes: []prefixEntry{
+			{IPV6Prefix: "ff00::/8"},
+			{IPV6Prefix: "fe00::/8"},
+		},
+	})
+	defer server.Close()
+
+	ips := newAWSIPs(serverIPRanges(server), time.Hour, nil)
+	AssertEqual(t, 0, len(ips.getCandidateNetworks(make([]byte, 17)))) // 17 bytes does not correspond to any net type
+	AssertEqual(t, 1, len(ips.getCandidateNetworks(make([]byte, 4))))  // netv4 networks
+	AssertEqual(t, 2, len(ips.getCandidateNetworks(make([]byte, 16)))) // netv6 networks
+}
+
+// RawMockHandler, like MockIPRangesHandler, sets
+// up a server to emulate aws. This handler
+// is configured by the json text, to facilitate testing
+// parsing.
+type RawMockHandler struct {
```

Comment:
No need for export here.

It might be easier to define this as a handler func and close over the required data for the test to create the handler for the instance:

```
var data []byte = generateSomehow()

handler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) { w.Write([]byte(m.data)) })
```